pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/equation/QuestionEvaluator.java

```diff
@@ -93,11 +95,13 @@ private void addValue(final QuestionType type, final String variable, final List
                                 .toList()));
                 return;
             case PICKLIST:
-                final var picklistValues = (List<Answer>) values;
                 this.values.put(variable, new EquationResponse(variable,
-                        StreamEx.of(picklistValues)
-                                .map(Answerable::getValue)
-                                .map(this::toString)
+                        StreamEx.of(values)
+                                .map(List.class::cast)
+                                .map(List::iterator)
+                                .map(Iterator::next)
+                                .map(SelectedPicklistOption.class::cast)
+                                .map(SelectedPicklistOption::getStableId)
```

Comment:
The `Iterator::next` method immediately reads as unusual to me. Can we add a comment here details what & why the `Iterator::next` is here?

Reply:
Sure. Maybe it's even better to break down the code into several small methods with natural names.